### PR TITLE
Add trace to web socket JSON/RPC responses

### PR DESCRIPTION
**Pull request by matthew1001**
## PR description
Trace WS JSON/RPC responses (successes and failures)

## Fixed Issue(s)
Fixes https://github.com/hyperledger/besu/issues/6276

## Example trace output

Successful `eth_getBlockByNumber`:

```
2023-12-12 10:43:41.553+00:00 | vert.x-eventloop-thread-1 | TRACE | WebSocketMessageHandler | {"jsonrpc":"2.0","id":51,"result":{"number":"0x0","hash":"0x74f92c171677a9f8b1eac53fc4f2c2e0d6bbc7a7b9b259fd0067f7b573951e16","mixHash":"0x63746963616c2062797a616e74696e65206661756c7420746f6c6572616e6365","parentHash":"0x0000000000000000000000000000000000000000000000000000000000000000","nonce":"0x0000000000000000","sha3Uncles":"0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347","logsBloom":"0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000","transactionsRoot":"0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421","stateRoot":"0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421","receiptsRoot":"0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421","miner":"0x0000000000000000000000000000000000000000","difficulty":"0x1","totalDifficulty":"0x1","extraData":"0xf88fa00000000000000000000000000000000000000000000000000000000000000000f869945c9437de876531c73c9618d13872f7508869f2539474474cf92e9b74df7a2ce2c4fbc0830c773b7a769462f241db855d439cc79f5d2bd4a58eeb9c5d01cd946d544f03911b74425dd4112e95f93b82ac6302f19477fd5cb2aea0b91e7e4faf7625bf9dbaee39ee47c080c0","size":"0x292","gasLimit":"0x1fffffffffffff","gasUsed":"0x0","timestamp":"0x58ee40ba","uncles":[],"transactions":[]}}

```

Failed `eth_getBlockByNumber`:

```
2023-12-12 10:44:02.881+00:00 | vert.x-eventloop-thread-1 | TRACE | WebSocketMessageHandler | {"jsonrpc":"2.0","id":51,"error":{"code":-32602,"message":"Invalid params"}}
```

### CODE REVIEW

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/worldview/BonsaiWorldState.java [hunk 1]
```diff
@@ -282,7 +282,10 @@ private void updateAccountStorageState(
           worldStateUpdater.getAccountsToUpdate().get(updatedAddress);
       final BonsaiAccount accountOriginal = accountValue.getPrior();
       final Hash storageRoot =
-          (accountOriginal == null) ? Hash.EMPTY_TRIE_HASH : accountOriginal.getStorageRoot();
+          (accountOriginal == null
+                  || worldStateUpdater.getStorageToClear().contains(updatedAddress))
+              ? Hash.EMPTY_TRIE_HASH
+              : accountOriginal.getStorageRoot();
```
Comment (macfarla) [inline ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/worldview/BonsaiWorldState.java:288]: @matthew1001 is this change related?

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/worldview/BonsaiWorldStateUpdateAccumulator.java [hunk 2]
```diff
@@ -328,17 +328,6 @@ public void commit() {
...
-                      .getAllAccountStorage(updatedAddress, updatedAccount.getStorageRoot())
-                      .forEach(
-                          (keyHash, entryValue) -> {
-                            final StorageSlotKey storageSlotKey =
-                                new StorageSlotKey(Hash.wrap(keyHash), Optional.empty());
-                            final UInt256 value = UInt256.fromBytes(RLP.decodeOne(entryValue));
-                            pendingStorageUpdates.put(
-                                storageSlotKey, new BonsaiValue<>(value, null, true));
-                          });
-                  updatedAccount.setStorageRoot(Hash.EMPTY_TRIE_HASH);
```
Comment (macfarla) [inline ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/worldview/BonsaiWorldStateUpdateAccumulator.java:341]: @matthew1001 is this change related?

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/websocket/WebSocketMessageHandler.java [hunk 3]
```diff
@@ -172,4 +178,12 @@ private void replyToClient(final ServerWebSocket websocket, final Object result)
   private JsonRpcResponse errorResponse(final Object id, final RpcErrorType error) {
     return new JsonRpcErrorResponse(id, error);
   }
+
+  private void traceResponse(final Object response) {
+    try {
+      LOG.trace(jsonObjectMapper.writeValueAsString(response));
+    } catch (JsonProcessingException e) {
+      throw new RuntimeException(e);
```
Comment (macfarla) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/websocket/WebSocketMessageHandler.java:186]: do we want to throw if we can't log? where does this exception come out?

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/worldview/BonsaiWorldState.java (hunk 1 above)
Comment (matthew1001) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/worldview/BonsaiWorldState.java:288]: Ummm - no - I'm confused as to where that's come in from. I looked at the files changed once I raised the PR and I don't think that was in there. But I can't explain how it did get there. I'll tidy up the branch in the morning to remove that commit (https://github.com/hyperledger/besu/pull/6285/commits/e90b4f1471de06935824db301f91c0a7222d3b1b) from the branch.

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/worldview/BonsaiWorldStateUpdateAccumulator.java (hunk 2 above)
Comment (matthew1001) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/worldview/BonsaiWorldStateUpdateAccumulator.java]: I don't know what happened with my branch but this commit has now been removed.

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/worldview/BonsaiWorldState.java (hunk 1 above)
Comment (matthew1001) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/worldview/BonsaiWorldState.java]: That commit has now been removed

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/websocket/WebSocketMessageHandler.java (hunk 3 above)
Comment (matthew1001) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/websocket/WebSocketMessageHandler.java]: Yeah I think you're right, logging it is safer. We do have places where we throw a runtime exception for the same JSON-parsing issue e.g. https://github.com/hyperledger/besu/blob/7770f966c92c638b7a6233a22d3fdd4f7a27f1ca/ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineExchangeTransitionConfiguration.java#L80

Perhaps that should be a log as well, instead of a runtime exception?

**Review by macfarla — APPROVED** [review]

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/websocket/WebSocketMessageHandler.java (hunk 3 above)
Comment (macfarla) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/websocket/WebSocketMessageHandler.java]: yeah I'd say the same thing for there
